src/client/Microsoft.Identity.Client/ManagedIdentity/V2/ImdsV2ManagedIdentitySource.cs

```diff
@@ -199,10 +204,29 @@ private async Task<CertificateRequestResponse> ExecuteCertificateRequestAsync(st
                 { OAuth2Header.XMsCorrelationId, _requestContext.CorrelationId.ToString() }
             };
 
+            // Normalize endpoint
+            Uri normalizedEndpoint = NormalizeAttestationEndpoint(attestationEndpoint, _requestContext.Logger);
+            if (normalizedEndpoint == null)
+            {
+                throw MsalServiceExceptionFactory.CreateManagedIdentityException(
+                    "attestation_endpoint_invalid",
+                    $"[ImdsV2] Attestation endpoint '{attestationEndpoint}' is invalid or unsupported.",
+                    null, ManagedIdentitySource.ImdsV2, null);
```

Comment:
We have a null check already in `CsrMetadata.cs`

Reply:
removed